training/deepspeech_training/util/audio.py

```diff
@@ -579,15 +579,23 @@ def get_dtype(audio_format):
 
 
 def pcm_to_np(pcm_data, audio_format=DEFAULT_FORMAT):
+    """
+    Converts PCM data (e.g. read from a wavfile) into a mono numpy column vector
```

Comment:
While I get the rationale, I'm unsure about the implementation for converting multi-channels here, and I wonder what kind of artifact we might be generating compared to relying on dedicated / external tooling (like sox)

Reply:
I proposed doing it using sox in #3512, but I see the counterarguments against. that.

I get the worry for artifacts with sample rate (which Reuben handled with an augmentation), but for channel merging, I'm pretty sure that averaging is exactly what sox would do. If you average channels for a low bitrate, it could maaayyybbeee make sense to convert to a higher bitrarte and do the averaging there with more floating point precision, then convert back down, but I doubt that it makes much of a difference with the kind of stuff that happens downstream here with MFCCs. I just tried to see if we get any meaningful difference in CTCs with this code and it seems like the answer is no up to 1e-7 at least:

```
# Produce CTCs
audiopath1 = '9780823435647-1009_16.wav' # mono version of 'I will give them some carrots'
sample1 = load_sample(audiopath1)
ctcs1 = model.generate_ctc_probs(sample1, session) # my wrapper producing CTCs for a single file
audiopath2 = 'stereo.wav' # stereo version of the same file
sample2 = load_sample(audiopath2)
ctcs2 = model.generate_ctc_probs(sample2, session)

# Compare
print("Max element-wise CTC difference: %f" % np.max(np.abs(ctcs1-ctcs2)))
print("Same? %s" % np.all(np.isclose(ctcs1, ctcs2, rtol=1e-07)))
```

Produces
```
Max element-wise CTC difference: 0.000000
Same? True
```

Reply:
Makes sense, but I'll let @reuben decide on this part